lib/util/UserInteractor.js

```diff
@@ -439,36 +441,98 @@ class UserInteractor {
+            if (value && value >= 1 && value <= choices.length) {
+                return value - 1;
+            }
+            return -1;
+        };
+        return UserInteractor._processChoice(
+            message,
+            question,
+            prompt,
+            valueValidator,
+            choicesHandlers,
+            handlersParams,
+            false,
+            0);
+    }
+
+    static _processChoice(message, question, prompt, valueValidator, 
+        choicesHandlers, handlersParams, confirmed, confirmedHandlerIndex, printMsg = true) {
         if (confirmed) {
-            return UserInteractor._callChoiceHandler(continueChoiceHandler, continueChoiceHandlerParams);
+            return UserInteractor._callChoiceHandler(choicesHandlers[confirmedHandlerIndex], handlersParams);
         }
         else {
-            return new Promise((resolve, reject) => {
-                if (printMsg) {
-                    if (message) {
-                        UserInteractor.printMessage(message);
-                    }
-                    UserInteractor.printMessage(UserInteractor.MESSAGES.CHOICES_CONTINUE_QUESTION);
+            if (printMsg) {
+                if (message) {
+                    UserInteractor.printMessage(message);
                 }
-                Prompt(UserInteractor.MESSAGES.CHOICES_PROMPT, function(value) {
+                UserInteractor.printMessage(question);
+            }
+            return UserInteractor.prompt(prompt).
+                then((value) => {
                     UserInteractor.printMessage('');
-                    const val = value.trim().toLowerCase();
-                    if (val === 'y') {
-                        resolve(UserInteractor._callChoiceHandler(continueChoiceHandler, continueChoiceHandlerParams));
-                    }
-                    else if (val === 'n') {
-                        resolve(UserInteractor._cancelOperation());
+                    const choiceIndex = valueValidator(value);
+                    if (choiceIndex >= 0) {
+                        return UserInteractor._callChoiceHandler(choicesHandlers[choiceIndex], handlersParams);
                     }
                     else {
                         UserInteractor.printMessage(UserInteractor.MESSAGES.CHOICES_INVALID_VALUE);
-                        resolve(UserInteractor.processCancelContinueChoice(
-                            message, continueChoiceHandler, continueChoiceHandlerParams, confirmed, false));
+                        return UserInteractor._processChoice(message, question, prompt, valueValidator, 
+                            choicesHandlers, handlersParams, confirmed, confirmedHandlerIndex, false);
                     }
                 });
-            });
         }
     }
 
+    static prompt(message, isPassword = false) {
+        return new Promise((resolve, reject) => {
+            let value = Prompt(message, null, isPassword ? { echo : '*' } : null);
```

Comment:
Can fix the following two critical issues:

1. print no `*` symbols even for passwords
2. never show the login key in clear as well